examples/differential_ik_whole_body.py

```diff
@@ -0,0 +1,92 @@
+"""
+This example shows PyRoboPlan capabilities for whole-body inverse kinematics (IK),
+using a 6-DOF arm on a mobile base capable of planar (3-DOF) motion.
+
+IK defines the task of finding a set of joint positions for a robot model to
+achieve a desired target pose for a specific coordinate frame.
+"""
+
+from pinocchio.visualize import MeshcatVisualizer
+import numpy as np
+
+from pyroboplan.core.utils import (
+    get_random_collision_free_state,
+    get_random_collision_free_transform,
+)
+from pyroboplan.ik.differential_ik import DifferentialIk, DifferentialIkOptions
+from pyroboplan.ik.nullspace_components import (
+    joint_limit_nullspace_component,
+    collision_avoidance_nullspace_component,
+)
+from pyroboplan.models.ur5 import (
+    load_ur5_on_base_models,
+    add_ur5_on_base_self_collisions,
+)
+
+
+if __name__ == "__main__":
+    # Create models and data
+    model, collision_model, visual_model = load_ur5_on_base_models()
+    add_ur5_on_base_self_collisions(model, collision_model)
+
+    data = model.createData()
+    collision_data = collision_model.createData()
+
+    target_frame = "tool0"
+
+    # Initialize visualizer
+    viz = MeshcatVisualizer(model, collision_model, visual_model, data=data)
+    viz.initViewer(open=True)
+    viz.loadViewerModel()
+    np.set_printoptions(precision=3)
+
+    # Set up the IK solver.
+    # The first 3 values in the weights are the base, and the other 6 are the arm.
+    # This means we are weighing base motion more than arm motion.
```

Comment:
Just to be clear, higher weight -> increased "cost" for moving that joint -> less overall motion is expected from the joint?

Reply:
Yep!